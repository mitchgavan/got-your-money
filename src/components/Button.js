import styled from 'styled-components'
import { space, width, fontSize, color } from 'styled-system'

export default styled.button`
  display: inline-block;
  vertical-align: middle;
  margin: 0 0 1rem 0;
  padding: 0.85em 1em;
  font-family: inherit;
  border: 1px solid transparent;
  border-radius: 100em;
  font-size: 0.9rem;
  line-height: 1;
  text-align: center;
  background-color: #1779ba;
  color: #fefefe;
  cursor: pointer;
  transition-property: background-color, color;
  transition-duration: 0.25s;
  transition-timing-function: ease-out;
  -webkit-appearance: none;
  ${space}
  ${width}
  ${fontSize}
  ${color}
`
