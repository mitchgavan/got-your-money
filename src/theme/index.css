body {
  font-family: 'Roboto', sans-serif;
  background-color: #f3f4f5;
}
