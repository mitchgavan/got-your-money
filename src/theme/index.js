export default {
  breakpoints: [
    32, 48, 64
  ],
  space: [
    0, 6, 12, 18, 24
  ],
  fontSizes: [
    12, 16, 18, 24, 36, 72
  ],
  colors: {
    black: '#111',
    blue: '#07c',
    green: '#3adb76'
  }
}
