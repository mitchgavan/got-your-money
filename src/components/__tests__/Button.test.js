import React from 'react'
import Button from '../Block'
import { shallow } from 'enzyme'
import 'jest-styled-components'
import toJson from 'enzyme-to-json'
import theme from '../../theme'

let component

beforeEach(() => {
  component = shallow(
    <Button
      theme={theme}
      color="white"
      bg="blue"
      p={2}
      fontSize={3}
      textAlign="center"
      width={200}
    />
  )
})

it('renders correctly', () => {
  expect(toJson(component)).toMatchSnapshot()
})

test('it applies styles according to passed props', () => {
  expect(component).toHaveStyleRule('color', theme.colors.white)
  expect(component).toHaveStyleRule('background-color', theme.colors.blue)
  expect(component).toHaveStyleRule('padding', theme.space[2] + 'px')
  expect(component).toHaveStyleRule('font-size', theme.fontSizes[3] + 'px')
  expect(component).toHaveStyleRule('width', '200px')
})
