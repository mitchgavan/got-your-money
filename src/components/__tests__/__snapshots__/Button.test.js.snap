// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  padding: 12px;
  width: 200px;
  font-size: 18px;
  color: #fefefe;
  background-color: #1779ba;
  text-align: center;
}

<div
  className="c0"
  color="white"
  fontSize={3}
  width={200}
/>
`;
